### PR TITLE
initial library

### PR DESCRIPTION
**Pull request by zsluedem**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    const resp = await this.deployService.getBlock({ hash: blockHash });
+    return resp;
+  }
+
+  async deploy(key, term, phloPrice, phloLimit, validAfterBlockNo, timestamp) {
+    const signedDeploy = signDeploy(key, {
+      term: term,
+      phloprice: phloPrice,
+      phloLimit: phloLimit,
+      vavalidafterblocknumber: validAfterBlockNo,
```
Comment (tgrospic) [inline lib/client.js:52]: Fields are lowercase. Type in `validafterblocknumber`.

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
```
Comment (tgrospic) [inline yarn.lock:4]: Is this file needed for something. I'm more in favor of npm only config.

File: examples/grpc_api_examples.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    Date.now()
+  );
+  console.log(deploy2);
+}
+
+(async () => {
+  await main();
+})().catch((e) => {
+  console.log(e);
+});
```
Comment (tgrospic) [inline examples/grpc_api_examples.js:71]: `await` can be caught with traditional try/catch.
```js
try {
  await main()
} catch (err) {
  console.log(err)
}
```

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.vscode
+.history
+node_modules
```
Comment (tgrospic) [inline .gitignore:3]: Please also ignore generated `rnode-grpc-gen` folder.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "nodejs-rchain",
+  "version": "0.0.1",
+  "description": "nodejs rchain lib",
+  "main": "index.js",
+  "repository": "https://github.com/zsluedem/nodejs-rchain.git",
+  "author": "Will",
+  "license": "MIT",
```
Comment (tgrospic) [inline package.json:8]: Please also include start scripts to generate JS binding files and hook to do it after install.
```json
"scripts": {
  "rnode-generate": "rnode-grpc --rnode-version v0.9.24",
  "postinstall": "npm run rnode-generate"
}
```

Comment (zsluedem) [reply package.json:8]: I am not acutually sure how to do this in the right way.

Comment (tgrospic) [reply package.json:8]: `npm install` downloads dependencies in `node_modules` and with `postinstall` hook it will also generate `rnode-grpc-gen` folder with `rnode-generate` script command.

**Comment by zsluedem** [comment]
@tgrospic updated!

File: lib/client.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
+const ec = require("elliptic");
+const { rnodeDeploy, signDeploy } = require("@tgrospic/rnode-grpc-js");
+const protoSchema = require("./rnode-grpc-gen/js/pbjs_generated.json");
+const grpc = require("@grpc/grpc-js");
+require("../lib/rnode-grpc-gen/js/DeployServiceV1_pb");
```
Comment (tgrospic) [inline lib/client.js:5]: This is not necessary because proto generated JS is initialized in `./lib/proto.js` file and loaded in `index.js`.

File: examples/sign_and_verify_example.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+const { signDeploy, verifyDeploy } = require("@tgrospic/rnode-grpc-js");
+require("..");
+const key = secp256k1.genKeyPair();
+// const key = '1bf36a3d89c27ddef7955684b97667c75454317d8964528e57b2308947b250b0'
+const sampleRholangCode = "Nil";
+const deployData = {
+  term: sampleRholangCode,
+  phloprice: 1,
+  phlolimit: 10e3,
+  validafterblocknumber: 0,
```
Comment (tgrospic) [inline examples/sign_and_verify_example.js:12]: Please add `timestamp` as `Date.now()`. I'll remove default in the future versions of _rnode-grpc-js_, because it's global dependency on Date object.

**Comment by zsluedem** [comment]
@tgrospic

File: README.md [hunk 8]
```diff
@@ -1,2 +1,17 @@
 # exchange-api-njs
-Nodejs-based api for exchanges
+
+## Introduction
+Nodejs-based api RChain RNode RPC.
+
+## Install
+
+> npm install --save git+https://git@github.com/rchain/exchange-api-njs.git
```
Comment (tgrospic) [inline README.md:8]: To install from GitHub there is also a special syntax.
`npm install <githubname>/<githubrepo>[#<commit-ish>]`

Also peer dependencies must be installed first for generator to work.
```sh
npm install --save-dev grpc-tools protobufjs

npm install rchain/exchange-api-njs
```
The problem is also with the current version of `rnode-grpc-js` which fails to find npm _.bin_ folder unless it's a direct dependency.
I fixed the issue in `v0.8.4` version so please update it in the PR.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,17 @@
...
+
+## Examples
+The features below are provided.
+
+1. [generate private key and public key](./examples/keys_example.js)
+2. [sign a deploy with the private key](./examples/sign_verify_examples.js)
+3. [use grpc api to interact with rnode](./examples/grpc_api_example.js)
+4. [Vault Api of rchain to do transfer and check balance](./examples/vault_example.js)
+5. [Get transaction info from block event data](./examples/transaction_example.js)
```
[the remark this answers is not in the document]
Comment (zsluedem) [reply README.md:17]: It is supposed to be when we merge the pr.

[the remark this answers is not in the document]
Comment (tgrospic) [reply README.md:18]: You can check the links on your repo. Only the last one is working.
<https://github.com/zsluedem/exchange-api-njs#examples>

File: README.md [hunk 10]
```diff
@@ -1,2 +1,18 @@
 # exchange-api-njs
-Nodejs-based api for exchanges
+
+## Introduction
+Nodejs-based api RChain RNode RPC.
+
+## Install
+
+> npm install --save-dev grpc-tools protobufjs
+> npm install rchain/exchange-api-njs
```
Comment (tgrospic) [inline README.md:9]: This should be in a code block because quotes are displayed in a single line.
